src/boostvqe/training_utils.py

```diff
@@ -319,3 +321,38 @@ def parameter_shift(
 
     circuit.set_parameters(original)
     return float(generator_eigenval * (forward - backward))
+
+
+def quantum_derivative_via_shifts(
```

Comment:
No, this is implemented here from scratch! It is just a function to say: if my gate is a rotation do the old psr, otherwise if it is RBS do this combination.